app/models/order_detail.rb

```diff
@@ -771,6 +747,34 @@ def problem_order?
     problem
   end
 
+  def can_dispute?
+    in_review? && dispute_at.blank?
+  end
```

Comment:
My goal was to make some of them independent so you ideally don't have to go more than one method down (and also make sure I don't have any infinite loops), but also to make them at least somewhat intention revealing, even if it means some predicates are redundant through other method calls.

I just pushed up another spec with a scenario that fails without the `dispute_at.blank?`.